README.md

```diff
@@ -21,55 +21,144 @@ This module is currently only tested on:
+
+## Parameters
+
+### Installation parameters
+
+These parameters control how Vault gets installed
+
+
+* `install_method`: Support the values `repo` or `archive`.  When `repo` is set the module will attempt to install a package corresponding with the value of `package_name`, when `archive` the module will attempt to download and extract a zip file from the `download_url`, the extracted file will be placed in the `bin_dir` folder
+
+#### When `install_method` is `package`
+
+* `package_name`:  Name of the package to install, default: vault
+* `package_ensure`: Desired state of the package, default: installed
+
+#### When `install_method` is `archive`
+
+* `download_url`: URL to download the vault zip distribution from.  You can specify a local file on the server with a fully qualified pathname, or use `http`, `https`, `ftp` or `s3` based URI's.
+* `download_dir`: Path to download the zip file to, default: `/tmp`
+* `manage_download_dir`: Boolean, whether or not to create the download directory, default: `false`
+* `download_filename`: Filename to (temporarily) save the downloaded zip file, default: `vault.zip`
+
+### Setup parameters
+
+* `user`: Customise the user vault runs as, will also create the user unless `manage_user` is false.
+
+* `manage_user`: Whether or not the module should create the user.
+
+* `group`: Customise the group vault runs as, will also create the user unless `manage_group` is false.
+
+* `manage_group`: Whether or not the module should create the group.
+
+* `bin_dir`: Directory the vault executable will be installed in.
+
+* `config_dir`: Directory the vault configuration will be kept in.
+
+* `purge_config_dir`: Whether the `config_dir` should be purged before installing the
+  generated config.
+
+* `install method`: Support the values `repo` or `archive`.  When `repo` is set the module will attempt to install a package corresponding with the value of `package_name`, when `archive
+
+* `service_name`: Customise the name of the system service
+
+* `service_provider`: Customise the name of the system service provider; this also controls the init configuration files that are installed.
+
+* `service_options`: Extra argument to pass to `vault server`, as per: `vault server --help`
+
+* `num_procs`: Sets the GOMAXPROCS environment variable, to determine how many CPUs Vault can use. The official Vault Terraform install.sh script sets this to the output of ``nprocs``, with the comment, "Make sure to use all our CPUs, because Vault can block a scheduler thread". Default: number of CPUs on the system, retrieved from the ``processorcount`` fact.
+
+
+### Configuration parameters
+
+By default, with no parameters the module will configure vault with some sensible defaults to get you running, the following parameters may be specified to configure Vault.  Please see [The official documentation](https://www.vaultproject.io/docs/config/) for further details of acceptable parameter values.
```

Comment:
My original reasoning for making `config_hash` one big hash was to reduce the maintenance burden of having to track new and removed options from vault. It also means a user can use an older or RC release of vault where the configuration is different to what this module supports.